File: src/main/java/com/deflatedpickle/heartdrops/events/FMLEventHandler.java
package com.deflatedpickle.heartdrops.events;

import net.minecraft.item.ItemStack;
import net.minecraftforge.event.entity.player.EntityItemPickupEvent;
import net.minecraftforge.fml.common.eventhandler.SubscribeEvent;

public class FMLEventHandler {
    @SubscribeEvent
    public void onEntityItemPickupEvent(EntityItemPickupEvent event) {
        ItemStack item = event.getItem().getItem();
        if (item.getUnlocalizedName().equals("item.heartdrops:heart")) {
            item.shrink(1);
        }
    }
}

File: src/main/scala/com/deflatedpickle/heartdrops/events/FMLEventHandler.scala
package com.deflatedpickle.heartdrops.events

import net.minecraftforge.event.entity.player.EntityItemPickupEvent
import net.minecraftforge.fml.common.eventhandler.SubscribeEvent

class FMLEventHandler {
  @SubscribeEvent
  def onEntityItemPickupEvent(event: EntityItemPickupEvent): Unit = {
    val item = event.getItem.getItem
    if (item.getUnlocalizedName == "item.heartdrops:heart") {
      event.getEntityPlayer.heal(2f)
      item.shrink(1)
    }
  }
}
